docs/rules/named-order.md

```diff
@@ -0,0 +1,68 @@
+# import/named-order
+
+Enforce a convention in the order of `import` / `export` / `require()` specifiers
++(fixable) The `--fix` option on the [command line] automatically fixes problems reported by this rule.
+
+## Rule Details
+
+The following patterns are valid:
+
+```js
+import { Alpha, Bravo } from 'foo'
+```
+
+```js
+const { Alpha, Bravo } = require('foo')
+```
+
+```js
+const Alpha = 'A'
+const Bravo = 'B'
+
+export { Alpha, Bravo }
+```
+
+The following patterns are invalid:
+
+```js
+import { Bravo, Alpha } from 'foo' // <- reported
+```
+
+```js
+const { Bravo, Alpha } = require('foo') // <- reported
+```
+
+```js
+const Alpha = 'A'
+const Bravo = 'B'
+
+export { Alpha, Bravo } // <- reported
+```
+
+### Options
+
+#### `esmodule`
+This option enables reporting of errors if `import` / `export` specifiers are not sorted. Its value is `true` by default.
+
+#### `commonjs`
+This option enablee reporting of errors if `require` specifiers are not sorted. Its value is `true` by default.
+
+#### `caseInsensitive`
+There is a `caseInsensitive` option available to be case insensitive. Its value is `false` by default.
+
+Without `caseInsensitive` option:
+```js
+import { alpha, Alpha, ALPHA } from 'foo' // <- reported, should sort as { ALPHA, Alpha, alpha }
+```
+
+With `caseInsensitive` option:
+```js
+import { alpha, Alpha, ALPHA } from 'foo' // <- will not report
+```
+
+#### `order`
```

Comment:
Shall we adopt https://palantir.github.io/tslint/rules/ordered-imports?

Reply:
Thanks to you, I realized the weirdness in some cases. I think it should consider the options below.

* "case-insensitive': Correct order is "Bar", "baz", "Foo". (This is the default.)
* "lowercase-first": Correct order is "baz", "Bar", "Foo".
* "lowercase-last": Correct order is "Bar", "Foo", "baz".

Reply:
@JounQin, Applied new order option to here 41b588839b8d7185c52d1f48f0b5a38390f6a1cb
thank you :)